src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -264,10 +270,70 @@ private int LineBreaksAfter(SyntaxToken currentToken, SyntaxToken nextToken)
             return 0;
         }
 
+        private static bool All<TNode>(SyntaxList<TNode> items, Func<TNode, bool> predicate)
+            where TNode : SyntaxNode
+        {
+            bool result = true;
+            for (int i = 0; i < items.Count; i++)
+            {
+                result &= predicate(items[i]);
+            }
+
+            return result;
+        }
+
+        private static bool IsAutoAccessorList(SyntaxToken token)
+            => token.Parent is AccessorListSyntax accessorList &&
+                All(accessorList.Accessors, a => a.Body == null);
+
+        private static bool IsAutoAccessorList(SyntaxNode? node)
+            => node?.Parent is AccessorListSyntax accessorList &&
+                All(accessorList.Accessors, a => a.Body == null);
+
+        private static bool HasInitializer(BasePropertyDeclarationSyntax basePropertyDeclaration)
+            => basePropertyDeclaration is PropertyDeclarationSyntax property && property.Initializer != null;
+
+        private static bool IsAccessorListFollowedByInitializer(SyntaxToken token)
+            => token.Parent is AccessorListSyntax accessorList &&
+                accessorList.Parent is BasePropertyDeclarationSyntax basePropertyDeclarationSyntax &&
+                HasInitializer(basePropertyDeclarationSyntax);
+
+        private static int LineBreaksBeforeOpenBrace(SyntaxToken currentToken, SyntaxToken nextToken)
+        {
+            if (nextToken.Parent.IsKind(SyntaxKind.Interpolation) ||
+                nextToken.Parent is InitializerExpressionSyntax)
+            {
+                return 0;
+            }
+            else if (IsAutoAccessorList(nextToken))
+            {
+                // currentToken is the property name, nextToken is the open brace
```

Comment:
>currentToken is the property name [](start = 19, length = 33)

What about an event case? Can we get here for it?

Reply:
The event accessors cannot be auto-implemented and follows the normal cases (they will not be collapsed to a single line)

Reply:
> The event accessors cannot be auto-implemented and follows the normal cases (they will not be collapsed to a single line)

I didn't see any check that explicitly excludes events.

---
In reply to: [531020650](https://github.com/dotnet/roslyn/pull/49495#discussion_r531020650) [](ancestors = 531020650)

Reply:
Let's see the behavior on the tests and you will then tell me what changes could be needed